core/string/ustring.h

```diff
@@ -238,16 +354,10 @@ class String {
 	String &operator+=(const wchar_t *p_str);
 	String &operator+=(const char32_t *p_str);
 
```

Comment:
Should we have operator+(const StrRange)() as well? One of the more common things we see is "String()" + "static_string". We could avoid a copy there, as well as a strlen().

Reply:
Oh yeah, I suppose that's a good idea. I'll get on that.

Reply:
I was hoping to address all the non-constructor functions in future PRs. A lot of these functions are duplicated for no reason, to be able to accept both `String` and `const char *`, so trying to address them all here will just increase the barrier to merge. Better to proceed in small steps, I think. Thinking about it, `operator+=` is also probably better avoided for now, just to be able to move more quickly with merging this one.

Reply:
Alright, fair enough.